src/lib/modules/Patron/PatronBulkExtendLoans/actions.js

```diff
@@ -0,0 +1,59 @@
+import { loanApi } from '@api/loans';
+import { searchReady } from '@api/utils';
+import {
+  sendErrorNotification,
+  sendSuccessNotification,
+} from '@components/Notifications';
+import {
+  fetchPatronCurrentLoans,
+  IS_LOADING as currentLoansIsLoading,
+} from '../PatronCurrentLoans/actions';
+import _isEmpty from 'lodash/isEmpty';
+
+export const IS_LOADING = 'bulkLoanExtend/IS_LOADING';
+export const SUCCESS = 'bulkLoanExtend/SUCCESS';
+export const HAS_ERROR = 'bulkLoanExtend/HAS_ERROR';
+
+export const bulkLoanExtension = (patronPid) => {
+  return async (dispatch) => {
+    dispatch({
+      type: IS_LOADING,
+    });
+    try {
+      const response = await loanApi.bulkExtendLoans(patronPid);
+
+      const notExtendedMessage = ` ${response.data.not_extended_loans.length} 
+            loan(s) could not be extended. Consult the ongoing loans section or contact the library for more details.`;
+      let successMessage = `${response.data.extended_loans.length} 
+            loan(s) have been successfully extended.`;
+
+      if (!_isEmpty(response.data.extended_loans)) {
+        if (!_isEmpty(response.data.not_extended_loans)) {
+          successMessage += notExtendedMessage;
+        }
+        dispatch(sendSuccessNotification('Extension success', successMessage));
+        dispatch({
+          type: currentLoansIsLoading,
+        });
+        await searchReady();
+        dispatch(fetchPatronCurrentLoans(patronPid));
+      } else {
+        let msg = 'There was no loans eligible for an extension.';
```

Comment:
Doubt: is it `no loans` or `no loan`?
```suggestion
        let msg = 'There was no loan eligible for an extension.';
```

Reply:
I checked, both are correct